manifests/vhost.pp

```diff
@@ -47,19 +55,38 @@
     fail('You must provide a target_ path for the certs to certs::vhost.')
   }
 
-  $crt = "${name}.crt"
-  $key = "${name}.key"
+  $defaults = { ensure => file, notify => Service[$service] }
```

Comment:
I don’t see $defaults in use, we could use that and have the conditionals set the additional content, then have one set of resource declarations outside of it. Do you think that would be understandable, or does the duplication make it easier to read?